### DISCUSSION TITLE
[📚] Support for Android 13 context permissions

### DISCUSSION THREAD

**Discussion by MicaelRodrigues** [invertase/react-native-firebase#6282] [Ideas] [answered]
With the upcoming [release](https://developer.android.com/about/versions/13/overview#timeline) of `Android Tiramisu` apps will need to request permission to receive push notifications:

- https://developer.android.com/about/versions/13/changes/notification-permission

It will start working similarly to IOS, that shows a system dialog for user choice. So, as a suggestion `messaging().requestPermission` should open that system dialog if the user hasn't granted/rejected permssions yet.

**Comment by mikehardy** [comment]
Ah, true yes. We've been working on this over at notifee (https://github.com/invertase/notifee/pull/351) 
react-native-permissions does not support it quite yet either https://github.com/zoontek/react-native-permissions/issues/688

Going to create an issue from this for tracking, thanks for linking the release timeline, it appears we have about 2 months, but the earlier the better really

**Comment by bang9** [comment]
Do we have any plans to implement this?
Now it seems that `react-native-permissions` supports notification permissions!

**Reply by mikehardy** [reply]
No concrete plans here - I'm traveling on vacation at the moment so I won't have time for a while, and no one has proposed a PR yet.
I've collaborate on the Notifee and react-native-permissions changes a bit though, to support the same, and can only recommend to use react-native-permissions (or Notifee) to manage this in the meantime if it is a current need.

It will happen here when I have time in the Fall timeframe but likely not before

**Reply by bang9** [reply]
I see, have a nice and safe trip! 🚂

**Comment by mikehardy** [comment, marked as answer]
We recommend using react-native-permissions (for CLI users) or expo-notifications (for Expo folks) these days

Cheers
